### ISSUE TITLE
Builds with -O2 when CMAKE_BUILD_TYPE=Debug is passed

### ISSUE THREAD

**Issue by yurivict** [ospray/ospray#204]

**Comment by carsonbrownlee** [comment]
Debug seems to simply set "-g" and otherwise use defaults, which should be -O0.  It seems that components/ospcommon/cmake/ispc.cmake is explicitly setting -O2 for non release builds, however this should only be for ISPC code.  @jeffamstutz is this correct behavior?

**Comment by jeffamstutz** [comment]
Yeah, using '-O2' for ISPC code in Debug builds has been there since before I joined Intel 2 years ago...simple enough to change to match C++-side code.

Probably the best option would be to provide ```ISPC_FLAGS_DEBUG```, ```ISPC_FLAGS_RELEASE```, and ```ISPC_FLAGS_RELWITHDEBINFO``` CMake variables, which could be overridden by users who are so inclined.

**Comment by carsonbrownlee** [comment]
While this flag is only used for ISPC, I have added separate ISPC flags as Jeff suggested into devel branch 3244cc27.  This may take a bit to propagate into github.

**Comment by jeffamstutz** [comment]
Let's keep this open until it's on Github.
